tracer/src/Datadog.Trace.ClrProfiler.Native/calltarget_tokens.cpp

```diff
@@ -500,15 +503,15 @@ HRESULT CallTargetTokens::ModifyLocalSig(ILRewriter* reWriter, TypeSignature* me
     *callTargetReturnToken = callTargetReturn;
     if (returnSignatureType != nullptr)
     {
-        *returnValueIndex = newLocalsCount - 4;
+        *returnValueIndex = newLocalsCount - 4 - additionalLocalsCount;
     }
     else
     {
         *returnValueIndex = static_cast<ULONG>(ULONG_MAX);
     }
-    *exceptionIndex = newLocalsCount - 3;
-    *callTargetReturnIndex = newLocalsCount - 2;
-    *callTargetStateIndex = newLocalsCount - 1;
+    *exceptionIndex = newLocalsCount - 3 - additionalLocalsCount;
+    *callTargetReturnIndex = newLocalsCount - 2 - additionalLocalsCount;
+    *callTargetStateIndex = newLocalsCount - 1; // Must be the last local.
```

Comment:
It seems strange to me that the last index does not subtract `additionalLocalsCount`. Is this a special case that's handled in the live debugger code?

Reply:
The reasoning I forced 'calltarget state' to be the last is because the mechanism of knowing if the method we're currently rewriting has already been rewritten before is by checking the type of the last local in the locals signature. (`localVarSig`).
https://github.com/DataDog/dd-trace-dotnet/blob/02b0522227fe3047a5607832bbedc05eec223fe2/tracer/src/Datadog.Trace.ClrProfiler.Native/calltarget_tokens.cpp#L353-L367

I could've made it cleaner by placing all the 'additional locals' before the calltarget ones but that required me to refactor the `calltarget_tokens.cpp` quiet a bit (I can further elaborate on it if you'd like) and I ditched refactoring the `calltarget_tokens.cpp` since we lived in a slydog and I didn't want to introduce potential conflicts down the line.

Does that make sense?